lib/validation/validators/slackChannelName.ts

```diff
@@ -0,0 +1,25 @@
+import { isString } from '../../support/Utils'
+
+const SLACK_CHANNEL_NAME_MAX_LENGTH = 80 as const
+const SLACK_CHANNEL_NAME_INVALID_SUMBOLS = /[ ,.!@#$%^&*?(){}<>=+|・'`~/\\\[\]\"]/
+
+export interface SlackChannelNameOptions {
```

Comment:
I think it's better to add separate validation for these options if we want like `startsWith` and `endsWith`. These have nothing to do with slack name validation 👀 And wouldn't add so much value. It's easy as doing:

```ts
useV(data, {
  name: {
    startsWith: startsWith('x_'),
    slackChannelName: slackChannelName()
  }
})
```

Reply:
Ah sorry wasn't looking at code 😓 OK I you wanted this to calculate the max length.

But in that case I see no point separating it into `prefix` and `suffix`. It could just be 1 single option 👀  Maybe `subtractLength`?

Reply:
Ah, sorry, my implementation was not aligned with the options provided.
The validator should have checked `value.startsWith(options.prefix)` and `values.endsWith(options.suffix)`.

I provided the options so that we can validate Slack channel name with one rule.

Providing `startsWith` and `endsWith` rules itself would be beneficial for other cases, but providing `slackChannelName` as a wrapper of 4 rules (`startsWith`, `endsWith`, `maxLength`, (and `regExp`)) could make the usage clear? 🤔 

```ts
useV(data, {
  name: {
    startsWith: startsWith('x_'),
    slackChannelName: slackChannelName(2) // meaning of 2 looks implicit?
  }
})
```

Reply:
Hmmm I disagree with it is wrapping 4 rules. `maxLength` and `regExp` is specific to Slack. Only slack validation rule knows what those are.

But `startsWith` and `endsWith` has nothing to do with slack. If we add these, there would be nothing stopping us from adding other rules too. Why not add `required` option too? You know what I mean 😳 

And yeah I think `slackChannelName(2)` is hard to understand what this `2` is 😅  I'm not 100% happy with `subtractLength` but I think it's still better.

Reply:
When do we need the prefixes? 👀  (Like does x_ has some special meaning?)

Reply:
Yeah, we have a use case where we want user to input their preferred channel name but it has be prefixed with `x_`. So, we only accept max 80 - 2 characters in this case 👀

In ordinary app, I think using computed value or `maxLength` would be more straight forward. Something like:

```ts
const input = ref('')
const prefix = 'x_'

const channelName = computed(() => {
  return `${prefix}${input.value}`
})

useV({
  input,
  channelName
}, {
  channelName: {
    slackChannelName: slackChannelName()
  }
})

// or

useV(data, {
  channelName: {
    maxLength: maxLength(SLACK_CHANNEL_NAME_MAX_LENGTH - 2)
  }
})
```

But, this is going to be tough when building form from Json schema (which we do these days

Reply:
In the latter example, it is needed to check characters available for Slack channel name (rule name doesn't matter)?

```ts
useV(data, {
  channelName: {
    slackChannelChars: slackChannelChars(),
    maxLength: maxLength(SLACK_CHANNEL_NAME_MAX_LENGTH - 2)
  }
})
```

Reply:
Yes I was just omitting it.

Reply:
I see.
I just thought it should have been there because we are talking (at least partially) about which we should choose, one wrapper rule or separated single purpose rules.

Reply:
> I just thought it should have been there because we are talking (at least partially) about which we should choose, one wrapper rule or separated single purpose rules.

Sorry if I confused you... but I don't remember talking about adding prefix or suffix rule for this validation 🤔